spark/v3.3/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestDelete.java

```diff
@@ -510,6 +510,22 @@ public void testDeleteWithConditionOnNestedColumn() {
         "Should have expected rows", ImmutableList.of(), sql("SELECT id FROM %s", tableName));
   }
 
+  @Test
+  public void testDeleteWithFilterOnNestedColumn() {
+    createAndInitNestedColumnsTable();
+
+    sql("INSERT INTO TABLE %s VALUES (1, named_struct(\"c1\", 3, \"c2\", \"v1\"))", tableName);
+    sql("INSERT INTO TABLE %s VALUES (2, named_struct(\"c1\", 2, \"c2\", \"v2\"))", tableName);
+
+    sql("DELETE FROM %s WHERE complex.c1 = 3", tableName);
```

Comment:
Delete conditions in `testDeleteWithConditionOnNestedColumn` can not be push down, so added this UT.